airflow/utils/types.py

```diff
@@ -72,3 +72,18 @@ class EdgeInfoType(TypedDict):
     """Extra metadata that the DAG can store about an edge, usually generated from an EdgeModifier."""
 
     label: str | None
+
+
+class DagRunTriggeredByType(str, enum.Enum):
+    """Class with TriggeredBy types for DagRun."""
+
+    CLI = "cli"  # for the trigger subcommand for dag command in cli: airflow dags trigger
+    OPERATOR = "operator"  # for the TriggerDagRunOperator
+    REST_API = "rest_api"  # for triggering the DAG via RESTful API
+    UI = "ui"  # for clicking the `Trigger DAG` button
+    TEST = "test"  # for dag.test()
+    SCHEDULER = "scheduler"  # for scheduler
```

Comment:
This should probably be say `TIME` or `TIMETABLE` or something to distinguish from `DATASET`. Maybe something like `TIME_TRIGGERD` and `EVENT_TRIGGERED` would work too.

Reply:
Actually, I was thinking that the `SCHEDULER` value is for timetable related triggering and `DATASET` value is for Dataset related triggering. 
I saw [this image](https://airflow.apache.org/docs/apache-airflow/stable/_images/dataset-scheduled-dags.png) on the documentation and it states "Triggered by Datasets" so I was thinking the `DATASET` is a good value for this.

Reply:
@uranusjr ?

Reply:
Your interpretation is not wrong, but the problem is that both timetable-related and database-related triggering are ultimately done by the scheduler, so this name is confusing. Some parts of Airflow use “scheduler” to refer to (only) timetable-related scheduling because historically that was the only way to schedule things; datasets were introduced later. We don’t need to follow the naming. Even if we keep the `SCHEDULER` variable—I don’t think we should—a better comment is still needed to explain this does not cover dataset-triggered runs.

Reply:
Hi @uranusjr, I missed this comment, sorry for that. 
How about changing the line with the following
```python
    TIMETABLE = "timetable"  # for timetable scheduler
```
as you suggested at the beginning. I think this will be clear enough to distinguish timetable based triggering and dataset based triggering. WDYT?

Reply:
I think that’s fine. Except _timetable scheduler_ is awkward; your wording in the next sentence is much better.

Reply:
yes, you are right. How about:
```python
    TIMETABLE = "timetable"  # for timetable based triggering
```